### PR TITLE
Adds an action for setting up brew and installing from brewfile

### PR DESCRIPTION
**Pull request by mike-carey**
## Problem

We would like an action that sets up brew now that Github-hosted runners no longer set it up by default.

## Solution

Adds an action that sets up brew and installs dependencies from a brewfile.

## Notes

- Requires #9 
- Accommodates both linux and macos

## Usage

```yml
jobs:
  # ...
  steps:
    - name: 'Checkout'
      uses: actions/checkout@v3

    - name: 'Install dev dependencies'
      uses: shopsmart/github-actions/actions/setup-homebrew@v2
```

### CODE REVIEW

File: actions/setup-brew/action.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Configure Homebrew cache'
+      if: inputs.cache == 'true'
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/Library/Caches/Homebrew/*
+          ~/Library/Caches/Homebrew/downloads/*
```
Comment (mike-carey) [inline actions/setup-brew/action.yml:32]: Hmm, these cache paths are probably incorrect.  🤔

Comment (mike-carey) [reply actions/setup-brew/action.yml:32]: So this will work with MacOS, not necessarily linux though.

File: actions/setup-brew/action.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        key: brew-${{ hashFiles('Brewfile') }}
+        restore-keys: brew-
+
+    - name: 'Add brew path'
+      shell: bash
+      run: echo "/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin" >> $GITHUB_PATH
```
Comment (mike-carey) [inline actions/setup-brew/action.yml:38]: This also pertains directly to linux.

File: .github/actions/test-setup-homebrew/action.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+
+name: 'Test setup-brew'
+description: 'Runs validation against the setup-brew action'
```
Comment (mike-carey) [inline .github/actions/test-setup-homebrew/action.yml:4]:
```suggestion
name: 'Test setup-homebrew'
description: 'Runs validation against the setup-homebrew action'
```

**Review by wenga86 — APPROVED** [review]

**Pull request merged**